src/pyk/kast/inner.py

```diff
@@ -136,6 +136,20 @@ def ml_pred(self) -> KInner:
             ml_term = KApply('#And', [ml_term, _i])
         return ml_term
 
+    @property
+    def pred(self) -> KInner:
+        items = []
+        for k in self:
+            val = self[k]
+            if type(val) is not KVariable or val.name != KVariable(k).name:
+                items.append(KApply('_==K_', [KVariable(k), val]))
+        if len(items) == 0:
+            return KToken('true', 'Bool')
+        term = items[0]
+        for _i in items[1:]:
+            term = KApply('_andBool_', [term, _i])
+        return term
```

Comment:
Consider using `functools.reduce`:

```suggestion
        conjuncts = [
            KApply('_==K_', KVariable(name), val) for name, val in self.items()
            if type(val) is not KVariable or val.name != k
        ]
        if not items:
            return KToken('true', 'Bool')
        return reduce(KLabel('__andbool__'), conjuncts)
```

(Not tested.)

Reply:
Now using `reduce`.